### ISSUE TITLE
Can't get it running, options not being passed?

### ISSUE THREAD

**Issue by benjolabs** [nfroidure/gulp-iconfont#79]
I am having trouble getting this running in my gulp stream. This is what my task looks like:

```
function iconfont() {

    var stream = gulp.src(config.svg.src + '/icons/*.svg')
        .pipe(iconfont({
            fontName: 'myfont', // required
            appendUnicode: true, // recommended option
            formats: ['ttf', 'eot', 'woff'], // default, 'woff2' and 'svg' are available
            timestamp: runTimestamp, // recommended to get consistent builds when watching files
            normalize: true,
            fontHeight: 1001
        }))
        .on('glyphs', function(glyphs, options) {
            // CSS templating, e.g.
            console.log(glyphs, options);
        })
        .pipe(gulp.dest(config.fonts.public));

    return stream;
}

gulp.task('iconfont', iconfont);
```

When running my tasks I get this error message:
```
'iconfont' errored after 3.28 ms
Error in plugin 'svgicons2svgfont'
Message:
    Missing options.fontName
```

It looks like none of the options I am setting actually get passed.

**Comment by nfroidure** [comment]
Please specify the version you're using, your OS and a working example to reproduce the problem.

**Comment by dougfabris** [comment]
Hey @nfroidure first of all thanks for the amazing plugin! So I'm having the same problem like @benjolabs to generate iconFont, did you have any solution for this now? (I'm using node v10.16.0, npm v6.9.0 and the last plugin version)

**Comment by nfroidure** [comment]
@dougfabris same that for the above, without a reproducible way to do this, we won't be able to help.

If it was a general issue, we wouldn't have 2 feedback in 4 years, there is great chances that the project were misused here.

**Comment by dougfabris** [comment]
@nfroidure thanks for the answer. Well, I was using gulp 3 and the iconFont was working perfectly. Now I'm using gulp 4 and because of that I had to change the way to write the code (I don't know, maybe the problem is here) and when I run iconfont I have the error.

```
// Code in Gulp 3 - working perfecly
gulp.task('iconfont', function(){
  return gulp.src(['assets/icons/*.svg']) // Source folder containing the SVG images
    .pipe(iconfontCss({
      fontName: fontName, // The name that the generated font will have
      path: 'assets/sass/templates/_icons.scss', // The path to the template that will be used to create the SASS/LESS/CSS file
      targetPath: '../../sass/partials/_icons.scss', // The path where the file will be generated
      fontPath: '../../assets/fonts/icons/' // The path to the icon font file
    }))
    .pipe(iconfont({
      prependUnicode: false, // Recommended option 
      fontName: fontName, // Name of the font
      formats: ['ttf', 'eot', 'woff'], // The font file formats that will be created
      normalize: true,
      timestamp: runTimestamp // Recommended to get consistent builds when watching files
    }))
    .pipe(gulp.dest('dist/fonts/icons/'));
});
```

```
// Code in Gulp 4 - when runs, I get error
function iconfont(done) {
	var stream = src(['src/icons/*.svg']) // Source folder containing the SVG images
    .pipe(iconfontCss({
      fontName: fontName, // The name that the generated font will have
      path: 'src/sass/templates/_icons.scss', // The path to the template that will be used to create the SASS/LESS/CSS file
      targetPath: '../../sass/partials/_icons.scss', // The path where the file will be generated
      fontPath: '../../src/fonts/icons/' // The path to the icon font file
    }))
    .pipe(iconfont({
      fontName: fontName, // Name of the font
      prependUnicode: false, // Recommended option 
      formats: ['ttf', 'eot', 'woff'], // The font file formats that will be created
      normalize: true,
      timestamp: runTimestamp // Recommended to get consistent builds when watching files
    }))
    .pipe(dest('dist/fonts/icons/'))
    return stream;
    done();
}

task("iconfont", iconfont);
```
When run the plugin, this is the error:
```
 Error in plugin "svgicons2svgfont"
Message:
    Missing options.fontName
Details:
    domain: [object Object]
    domainThrown: true
```
Thanks for your attention!

**Comment by Chenzo** [comment]
Hey @dougfabris - I'm encountering the same problem too. This is an old issue, but maybe being caused by something new now.

I even reduced our function to be simple to reduce the number of failure points:

`function iconfont() {
    return gulp
        .src([paths.images.svg + '/*.svg'])
        .pipe(iconfont({
            fontName: "icons",
            prependUnicode: true,
            formats: ['ttf', 'eot', 'woff', 'woff2'],
            normalize: true,
            fontHeight: 1001
        }))
        .pipe(gulp.dest(paths.styles.dest + '/fonts/icons/'));
}`

Did you ever solve this, @dougfabris ?

**Comment by dougfabris** [comment]
Hey @Chenzo I not solved yet, the problem persists in plugin "svgicons2svgfont" when I try to run iconFont, I don't know what's happening... I believe that if we need use iconFont for now, we have to come back to gulp v3

**Comment by Chenzo** [comment]
@dougfabris - So, I just made a plain repo with nothing but this: https://github.com/Chenzo/gulp-svgs-to-font

My function `gulp iconfontnew` fails

But the example function from the gulp-iconfont works `gulp Iconfont`

we're doing something wrong, I think. (and this is in Gulp 4)

**Comment by dougfabris** [comment]
Cool about the repo mate! So.. yeah I agree with you that we can be doing something wrong when we write function in gulp 4, or maybe the plugin "svgicons2svgfont" are not able to understand the syntax. As I told to @nfroidure the plugin in gulp 3 works great.

**Comment by Chenzo** [comment]
Yeah, I'm just going to take that working function and modify it to my needs... I have no idea why that works and the other one doesn't 

¯\_(ツ)_/¯

Just wanted to let you know that it DOES work and you can get it going (mostly, I need to get it working so my devs/designers get off my back. lol.)

hopefully you can get yours working too, @dougfabris 
(And hopefully this helps someone else out there.)

**Comment by thomas-j-moffett** [comment]
Facepalm moment for me :facepalm:  ...but I ran into this "issue" as well when converting my gulpfile from Gulp 3.x to Gulp 4.x.

So, I was tediously taking my task names and making them functions to use rather than all anonymous functions. I had a task named `iconfont` (as a string), so I created a function named `iconfont`...not catching that my gulpfile already had `var iconfont = require('gulp-iconfont');` at the top...doh!

It took a little debugging for me to see that something very odd was being passed as the `options`, but thankfully, it didn't take too much time for me to see the duplicate names. I simply renamed my function to `iconfontTask`, and it worked fine.

Anyway, I hope my dumb moment can help someone else who runs into this issue!!
